vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/ContentBlockManager.java

```diff
@@ -315,6 +315,40 @@ public Integer findMaxContentOrder(String slideId) {
         return contentBlockRepository.findMaxContentOrder(slideId);
     }
 
+    /**
+     * Adjusting the content order of the blocks of slide once it is dragged and changed position
+     * 
+     *  @param blockId - id of block whose content order needs to be adjusted.
+     *  @param contentOrder - value with which contentOrder of block needs to be updated.
+     *  
+     */
+
+    @Override
+    public void adjustContentOrder(List<ContentBlock> contentBlockList) throws BlockDoesNotExistException {
+        
+        if(contentBlockList!=null)
+        {
+            List<ContentBlock> contentBlocks = new ArrayList<>();
+            for(ContentBlock eachBlock:contentBlockList )
+            {
+                String blockId = eachBlock.getId();
+                int contentOrder = eachBlock.getContentOrder();
+                if (blockId == null) { 
+                    return;
+                }
+                Optional<ContentBlock> contentBlock = contentBlockRepository.findById(blockId);
+                if (contentBlock.isPresent()) {
+                    ContentBlock contentBlockObj = contentBlock.get();
+                    contentBlockObj.setContentOrder(contentOrder);
+                    contentBlocks.add(contentBlockObj);  
+                } else {
+                    throw new BlockDoesNotExistException("Block Id not present");
```

Comment:
let's make sure the transaction is being rolled back in this case by adding a `rollbackFor` property to the `@Transactional` annotation.